### PR TITLE
Snaps documentation

### PR DESCRIPTION
**Pull request by rekmarks**
**This PR must not be merged until the `DO NOT MERGE` label is removed.**

Adds documentation for Snaps. The Snaps docs consist of:
- An introduction / tutorial (courtesy @mapachurro)
- API documentation
- A development guide (still WIP)

We can merge this PR although the development guide is still unfinished, and finish it in the days after the launch.

Supersedes #289

### CODE REVIEW

File: docs/.vuepress/config.js [hunk 1]
```diff
@@ -6,27 +6,46 @@ module.exports = (_ctx) => ({
     '/': {
       lang: 'en-US',
       title: 'MetaMask Docs',
-      description: 'Developer documentation for the MetaMask Ethereum wallet'
-    }
+      description: 'Developer documentation for the MetaMask Ethereum wallet',
```
Comment (rekmarks) [inline docs/.vuepress/config.js:9]: This file was updated to display the new pages in the sidebar, and while I was here I linted it with `prettier` and removed a useless function.

Comment (mapachurro) [reply docs/.vuepress/config.js:9]: Thank you!

**Review by BboyAkers — APPROVED** [review]
LGTM, Went ahead and merged it.
What are the "to do" items for post launch you'd like to see completed?

**Comment by rekmarks** [comment]
This was reverted.

**Pull request merged**